File: app/assets/javascripts/blocks/embed_block.js

SirTrevor.Blocks.Embed = (function() {
  return Radiator.Block.extend({
    title: function() { return "Embed"; },
    
    icon_name: "embed",
    
    type: "embed",

    metadata: function() {
      return this.text_field("url", "URL");
    },

  });
})();

File: app/models/blocks/embed_block.rb
module Blocks
  class EmbedBlock
    
    delegate :html, to: :embed
    def initialize block
      @block = block
    end
    
    def url
      @block["url"]
    end

    def embed
      @embed ||= OEmbed::Providers.get(url)
    end
  end
end
